### ISSUE TITLE
Can't login with argocd cli when using oidc sso

### ISSUE THREAD

**Issue by enderson-pan** [argoproj/argo-cd#4104]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

After configuring argocd to use oidc, I've successfully login with web-ui, however, failed using argocd cli. The argocd cli
may mistakenly set the redirect uri.

```
argocd login argocd.holytiny.com --sso

Opening browser for authentication
INFO[0001] RequestedClaims: map[groups:essential:true ]
Performing authorization_code flow login: https://gitlab.holytiny.com/oauth/authorize?access_type=offline&claims=%7B%22id_token%22%3A%7B%22groups%22%3A%7B%22essential%22%3Atrue%7D%7D%7D&client_id=9fc2a32f7640b193a76b213910d79042598259481cc600a46a211152b4432f96&redirect_uri=http%3A%2F%2Flocalhost%3A8085%2Fauth%2Fcallback&response_type=code&scope=openid+profile+email&state=KHWpnMPiGE
```
It seems argocd cli set the redirect_uir to localhost. I've read the doc of https://argoproj.github.io/argo-cd/operator-manual/argocd-cm.yaml, but cannot find where to set redirect_uir in oidc fields.

**To Reproduce**
1. Setup oidc in argocd-cm.yaml
2. Apply argocd-cm.yaml
3. Test web ui successfully login.
4. use argocd cli to login with --sso

**Version**
```shell
argocd: v1.6.0+c10ae24
  BuildDate: 2020-06-16T22:41:56Z
  GitCommit: c10ae246ab02f1356147118a1979fedcd1ceb704
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.6.1+159674e
  BuildDate: 2020-06-19T00:41:05Z
  GitCommit: 159674ee844a378fb98fe297006bf7b83a6e32d2
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:kustomize/v3.6.1 GitCommit:c97fa946d576eb6ed559f17f2ac43b3b5a8d5dbd BuildDate:2020-05-27T20:47:35Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
  Kubectl Version: v1.14.0
```

**Comment by voldemortensen** [comment]
`argocd ... --sso` opens a temporary HTTP server on localhost in order to receive the token sent back from the auth provider. Without returning to localhost `argocd` wouldn't be able to complete the OAuth2 process.

If you were expecting to be redirected to your Argo dashboard, `argocd login` is only used to authorize the `argocd` cli tool to use the API.

**Comment by enderson-pan** [comment]
Hi,@voldemortensen 
Thanks for reply. I'v got it.

**Comment by enderson-pan** [comment]
@StefanMadsen Hi, finally I found argocd maynot have nothing to do with this issue. This issue may be the lacking of function of the oidc provider. Oidc provider should provide some way to let the command line tool to complete the auth, just like what the github doing.

**Comment by StefanMadsen** [comment]
@enderson-pan i agree, we fixed it with azuread by allowing an additional "localhost" callback url as well as argos public server url
its definitly not a argocd cli issue.

**Comment by dannysauer** [comment]
Since this issue is one of the top Google search results but the "resolution" doesn't actually provide enough information for someone else to use... The CLI configuration is partially documented under the Okta setup section of the docs:
https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/okta/#cli-login

TLDR: allow `http://localhost:8085/auth/callback` as a second redirect URL.  The port changes if you specify an alternate `--sso-port` (this is documented on the keycloak setup page 🤦), but 8085 is the default.
